src/selectors/TraversalSelector.h

```diff
@@ -0,0 +1,158 @@
+/**
+ * @file TraversalSelector.h
+ * @author F. Gratl
+ * @date 11.06.18
+ */
+
+#pragma once
+
+#include <containers/cellPairTraversals/C08Traversal.h>
+#include <containers/cellPairTraversals/CellPairTraversal.h>
+#include <containers/cellPairTraversals/SlicedTraversal.h>
+#include <pairwiseFunctors/CellFunctor.h>
+#include <array>
+#include <vector>
+#include "utils/ExceptionHandler.h"
+#include "utils/Logger.h"
+
+namespace autopas {
+
+/**
+ * Provides a way to iterate over the possible choices of TraversalOption.
+ */
+static std::vector<TraversalOptions> allTraversalOptions = {TraversalOptions::c08, TraversalOptions::sliced};
+
+/**
+ * Selector for a container traversal.
+ * @tparam ParticleCell
+ */
+template <class ParticleCell>
+class TraversalSelector {
+ public:
+  /**
+   * Constructor of the TraversalSelector class.
+   * @param dims Array with the dimension lengths of the domain.
+   * @param allowedTraversalOptions Vector of traversals the selector can choose from.
+   */
+  TraversalSelector(const std::array<unsigned long, 3> &dims,
+                    const std::vector<TraversalOptions> &allowedTraversalOptions)
+      : _dims(dims), _allowedTraversalOptions(allowedTraversalOptions) {}
+
+  /**
+   * Gets the optimal traversal for a given cell functor. If no traversal is selected yet a optimum search is started.
+   * @tparam PairwiseFunctor The functor that defines the interaction of two particles.
+   * @tparam useSoA
+   * @tparam useNewton3
+   * @param pairwiseFunctor The functor that defines the interaction of two particles.
+   * @return Smartpointer to the optimal traversal.
+   */
+  template <class PairwiseFunctor, bool useSoA, bool useNewton3>
+  std::unique_ptr<CellPairTraversal<ParticleCell>> getOptimalTraversal(PairwiseFunctor &pairwiseFunctor);
+
+  /**
+   * Evaluates to optimal traversal based on a given cell functor.
```

Comment:
"has to reset particles" might also be here

Reply:
#71